### PR TITLE
Statistics generation enhancements 

### PR DESCRIPTION
**Pull request by anvmn**
#426

### CODE REVIEW

**Comment by anvmn** [comment]
@mariano-dagostino please review

File: server/hedley/modules/custom/hedley_stats/hedley_stats.module [hunk 1]
```diff
@@ -313,69 +319,136 @@ function hedley_stats_calculate_stats_worker($queue_item) {
...
-    $wrapper = entity_metadata_wrapper('node', $health_center_id);
-
-    // Get all clinics of type 'FBF' for the HC.
-    $fbf_clinics = hedley_health_center_get_clinics_by_health_center($health_center_id, 'fbf');
-    // Calculate statistics that are provided for FBF clinics only.
-    $completed_program = $missed_sessions = [];
-    if (!empty($fbf_clinics)) {
-      list($completed_program, $missed_sessions) = hedley_stats_get_session_attendance_stats_by_health_center($health_center_id, $fbf_clinics);
-    }
+function hedley_stats_pull_stats_for_health_center($health_center_id) {
```
Comment (mariano-dagostino) [inline server/hedley/modules/custom/hedley_stats/hedley_stats.module]: @anvmn I'm not sure if the logic defined here is correct. My understanding is if there is no stats for any of the following cache_data the function will return an empty array. Could be possible to have a null value for any of the following `hedley_stats_handle_cache` and still want to treat it as a valid value?

If this is not the case. The PR looks good.

Comment (anvmn) [reply server/hedley/modules/custom/hedley_stats/hedley_stats.module]: @mariano-dagostino 
Any of those returnung NULL means that HC statistics are being recalculated by server.
In this case, we return empty array, indicating that statistics are not available at current moment.

**Comment by anvmn** [comment]
@mariano-dagostino 
Made minor change at [82a1548](https://github.com/TIP-Global-Health/eheza-app/pull/427/commits/82a1548b9ac124b0f6e7a777a3ae38c66d5710f1).

**Review by mariano-dagostino — APPROVED** [review]

**Pull request merged**